### DISCUSSION TITLE
Does this interfere with anticheat such as Vanguard and VAC, anything with kernel anticheat?

### DISCUSSION THREAD

**Discussion by HyperLives** [cafali/SnapKey#18] [Q&A] [answered]
Since the rise of Snap Tap and SOCD everyone is wanting a piece including me, however the difference between those keyboards and software like yours is that the keyboard are ran on a firmware level on the keyboard which has been approved by developers already. Would the software like these be detected as cheating. We know it doesn't interfere with game files but it is still risky.

**Comment by minteeaa** [comment, marked as answer]
the system uses a driver supplied by the windows system, as a result some kernel level drivers can decide to block that driver in favor of getting raw keyboard data into the game, effectively disabling SnapKey. (CS:S has an implementation of this)

as you state this is not run on the hardware level and even if it doesn't interact with game files or inject a process into the game, companies like Riot may one day decide to block this specific application from launching, or outright make a statement that says it isn't allowed.

this being said, keep updated on the patch notes of games with kernel level anticheats. for the most part, companies are worried about the *idea* of Snap Tap, and not how it is run.

so far though, it has been VAC and Vanguard safe for about everyone i've asked. as long as you don't modify the source to add your own malicious code, you should be fine 👍
